atomics/wait/web-worker-agent/nan-timeout.htm

```diff
@@ -0,0 +1,29 @@
+<!doctype html>
+<meta charset=utf-8>
+<title></title>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script>
+
+/**
+Tests Atomics.wait in a web worker.  The worker thread waits on the
+SharedArrayBuffer slot 0 to with a NaN timeout.
+*/
+
+test(function(t) {
+    var ia = new Int32Array(new SharedArrayBuffer(Int32Array.BYTES_PER_ELEMENT));
+    var worker = new Worker("workers/nan-timeout.js");
+    window.setTimeout(() => Atomics.wake(ia, 0), 1000);
+    
+    worker.onerror = function(e) {
+      assert_false(true);
+    }
+    worker.onmessage = function(e) {
+      let val = e.data[0];
+      if (typeof val == "undefined") return;
```

Comment:
For an unknown reason, onmessage is called twice. In one of the calls `e.data[0]` is undefined. That's the call I skip.